src/Integration.UnitTests/Binding/CSharpVBBindingConfigProviderTests.cs

```diff
@@ -396,11 +424,23 @@ public CSharpVBBindingConfigProvider CreateTestSubject()
                     .Setup(x => x.Generate(bindingRootFolder, ExpectedProjectKey, language.FileSuffixAndExtension))
                     .Returns(FilePathResponse);
 
+                solutionBindingFilePathGeneratorMock
+                    .Setup(x => x.Generate(bindingRootFolder, ExpectedProjectKey, string.Empty))
+                    .Returns(AdditionalFilePathResponse);
+
+                var sonarProperties = PropertiesResponse.ToDictionary(p => p.Key, y => y.Value);
+
+                sonarLintConfigGeneratorMock = new Mock<ISonarLintConfigGenerator>();
+                sonarLintConfigGeneratorMock
+                    .Setup(x => x.Generate(It.IsAny<IEnumerable<SonarQubeRule>>(), sonarProperties, language))
```

Comment:
Is generating the `sonarProperties` dictionary here providing a stronger check than using `It.IsAny...`?

i.e. is Moq actually checking that the contents of the properties of the dictionaries are equivalent?

Reply:
Moq will fail unless you use It.IsAny or pass the exact dictionary it expects. So I think it makes more sense to pass sonarProperties that Any.

Reply:
It isn't the "exact dictionary" - it's a different object with equivalent contents, hence the question
> is Moq actually checking that the contents of the properties of the dictionaries are equivalent?
And apparently it is.